R/Gargle-class.R

```diff
@@ -208,12 +208,14 @@ Gargle2.0 <- R6::R6Class("Gargle2.0", inherit = httr::Token2.0, list(
   },
   #' @description (Attempt to) refresh a Gargle2.0 token
   refresh = function() {
-    cred <- get("refresh_oauth2.0", asNamespace("httr"))(
-      self$endpoint, self$app,
-      self$credentials, self$params$user_params, self$params$use_basic_auth
+    cred <- refresh_oauth2.0(
+      self$endpoint, self$app, self$credentials,
+      package = self$package
     )
     if (is.null(cred)) {
       token_remove_from_cache(self)
+      # TODO: return(NULL) here? why do we return the current, invalid,
+      # unrefreshed token?
```

Comment:
The more I think about this, I think, when a token fails to refresh, we should clear its refresh token. And possibly do more to indicate that it's invalid and why?

Long-term, httr2-wise, I think a token needs a way to communicate its viability back to a client package that may be managing it in memory.

Reply:
Yeah, seems like this should error as well as removing from the cache?